plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/tree/library/JarNode.java

```diff
@@ -50,41 +52,46 @@ public JarNode(
       @Assisted Path project,
       @Assisted NodeSettings nodeSettings,
       JavaResources javaResources,
-      JavaNavigationService service,
+      DtoFactory dtoFactory,
+      JavaLanguageExtensionServiceClient service,
       JavaNodeFactory nodeFactory) {
     super(jar, nodeSettings);
     this.project = project;
     this.javaResources = javaResources;
+    this.dtoFactory = dtoFactory;
     this.service = service;
     this.nodeFactory = nodeFactory;
   }
 
   @NotNull
   @Override
   protected Promise<List<Node>> getChildrenImpl() {
+    ExternalLibrariesParameters params = dtoFactory.createDto(ExternalLibrariesParameters.class);
+    params.setProjectUri(project.toString());
+    params.setNodeId(getData().getId());
     return service
-        .getLibraryChildren(project, getData().getId())
+        .libraryChildren(params)
         .then(
-            new Function<List<JarEntry>, List<Node>>() {
-              @Override
-              public List<Node> apply(List<JarEntry> entries) throws FunctionException {
-                List<Node> nodes = new ArrayList<>();
+            (Function<List<JarEntry>, List<Node>>)
```

Comment:
Do we need the cast here? Maybe ```(List<JarEntry> entries)-> {``` would be enough

Reply:
The cast `(Function<List<JarEntry>, List<Node>>)` is necessary here. Here I use Function not Operation.